govc/device/pci/add.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright (c) 2014-2016 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pci
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"strconv"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type add struct {
+	*flags.VirtualMachineFlag
+}
+
+func init() {
+	cli.Register("device.pci.add", &add{})
+}
+
+func (cmd *add) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.VirtualMachineFlag, ctx = flags.NewVirtualMachineFlag(ctx)
+	cmd.VirtualMachineFlag.Register(ctx, f)
+}
+
+func (cmd *add) Description() string {
+	return `Add PCI Passthrough device to VM.
+
+Examples:
+  govc device.pci.add -vm $vm {pci address}*
+  govc device.info -vm $vm`
+}
+
+func (cmd *add) Process(ctx context.Context) error {
+	if err := cmd.VirtualMachineFlag.Process(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *add) Run(ctx context.Context, f *flag.FlagSet) error {
+	if len(f.Args()) == 0 {
+		return errors.New("no pci device addresses are supplied")
+	}
+
+	reqDevices := map[string]*types.VirtualMachinePciPassthroughInfo{}
+	for _, n := range f.Args() {
+		reqDevices[n] = nil
+	}
+
+	vm, err := cmd.VirtualMachine()
+	if err != nil {
+		return err
+	}
+	if vm == nil {
+		return flag.ErrHelp
+	}
+
+	vmConfigOptions, err := vm.QueryConfigTarget(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, d := range vmConfigOptions.PciPassthrough {
+		info := d.GetVirtualMachinePciPassthroughInfo()
+		if info == nil {
+			return errors.New("received invalid pci passthrough info")
+		}
+
+		_, ok := reqDevices[info.PciDevice.Id]
+		if !ok {
+			continue
+		}
+		reqDevices[info.PciDevice.Id] = info
+	}
+
+	newDevices := []types.BaseVirtualDevice{}
+	for id, d := range reqDevices {
+		if d == nil {
+			return fmt.Errorf("%s is not found in allowed PCI passthrough device list", id)
+		}
+		device := &types.VirtualPCIPassthrough{}
+		device.Backing = &types.VirtualPCIPassthroughDeviceBackingInfo{
+			Id: d.PciDevice.Id, DeviceId: strconv.Itoa(int(d.PciDevice.DeviceId)),
+			SystemId: d.SystemId, VendorId: d.PciDevice.VendorId,
+		}
+		device.Connectable = &types.VirtualDeviceConnectInfo{StartConnected: true, Connected: true}
+		newDevices = append(newDevices, device)
+	}
+
+	if err := vm.AddDevice(ctx, newDevices...); err != nil {
+		return err
+	}
+	fmt.Printf("%v has been successly added to the VM\n", f.Args())
```

Comment:
Can you remove this and just describe in the help text that there is no output upon success. And maybe add to the examples how `device.info` can be used to show details of what was just added.